default.json5

```diff
@@ -82,13 +82,15 @@
       matchUpdateTypes: [
         "major",
       ],
+      matchCurrentVersion: "!/^0/",
```

Comment:
Hmm, I don't like this too much, because all of our `hivemq-commons` dependencies are 0.:
```
hivemq-configurationService = "0.11.2"
hivemq-customContainer = "0.5.2"
hivemq-logbackTestAppender = "0.4.0"
hivemq-rateLimiter = "0.2.1"
```
As well as some other external dependencies, e.g.:
```
zeroAllocationHashing = "0.26ea0"
```

I think this will lead to a lot of missing updates :(

Reply:
I think the collateral if this exclusion is too big:
```
concurrentUnit = "0.4.6"
defaults = { id = "io.github.sgtsilvio.gradle.defaults", version = "0.2.0" }
docker = { id = "com.palantir.docker", version = "0.36.0" }
graalvm-native = { id = "org.graalvm.buildtools.native", version = "0.10.2" }
gradleOci-junitJupiter = "0.5.0"
hivemq-batchAggregator = "0.2.2"
hivemq-configurationService = "0.11.2"
hivemq-configurationService = "0.9.1"
hivemq-customContainer = "0.5.2"
hivemq-logbackTestAppender = "0.4.0"
hivemq-rateLimiter = "0.2.1"
hivemq-result = "0.1.1"
jackson-databind-nullable = "0.2.6"
jacoco = "0.8.11"
jacoco = "0.8.7"
javadocLinks = { id = "io.github.sgtsilvio.gradle.javadoc-links", version = "0.8.0" }
javet-sanitizer = "0.2.0-antlr-4.13.1"
jmh = { id = "me.champeau.jmh", version = "0.7.2" }
jose4j = "0.9.4"
jose4j = "0.9.6"
license = { id = "com.github.hierynomus.license", version = "0.16.1" }
licenseReport = { id = "com.github.hierynomus.license-report", version = "0.16.1" }
metadata = { id = "io.github.sgtsilvio.gradle.metadata", version = "0.5.0" }
metadata = { id = "io.github.sgtsilvio.gradle.metadata", version = "0.6.0" }
mustache = "0.9.11"
oci = { id = "io.github.sgtsilvio.gradle.oci", version = "0.10.0" }
openJdk-jol = "0.17"
proguardGradlePlugin = "0.6.0"
proguard = { id = "io.github.sgtsilvio.gradle.proguard", version = "0.7.0" }
prometheus-simpleClient = "0.16.0"
reflections = "0.10.2"
vaadin-sassCompiler = "0.9.13"
versions = { id = "com.github.ben-manes.versions", version = "0.42.0" }
versions = { id = "com.github.ben-manes.versions", version = "0.45.0" }
zeroAllocationHashing = "0.16"
zeroAllocationHashing = "0.26ea0"
```

Reply:
Semantically any update to a 0.x version can be a breaking change, so should at least be treated as a major version update.
Regarding `hivemq-commons`, this is an issue of these libraries. They should not be auto-updated if they are still 0.x. Otherwise they should be made stable, so get a 1.x version.

Reply:
Why are we changing this? Did the current configuration cause any particular issue?

If we are going forward with this change, maybe adding this config might make it cleaner:
```
{
  "packageRules": [
    {
      matchCurrentVersion: "!/^0/",
      "enabled": false
    }
  ]
}
```

Reply:
> Why are we changing this? Did the current configuration cause any particular issue?

I implemented a breaking change in the Gradle OCI plugin (which has still a 0.x version, so it is still ok to make a breaking change).
This breaking change requires that the version is updated in all repositories at the same time. That's why it will be done "manually" with my changes.
Unfortunately, it was included in all renovate minor updates and caused confusion.
I made this change because this can happen with any library using 0.x versions at any time.

Reply:
- these are all build dependencies:
  - defaults
  - docker
  - graalvm-native
  - jacoco
  - javadocLinks
  - jmh
  - license
  - licenseReport
  - metadata
  - oci
  - proguard
  - vaadin-sassCompiler -> last update 2015, vaadin will not be updated => not an issue
  - versions
- hivemq-commons -> owned by hivemq, either stabilize API or update manually
  - hivemq-batchAggregator
  - hivemq-configurationService
  - hivemq-configurationService
  - hivemq-customContainer
  - hivemq-logbackTestAppender
  - hivemq-rateLimiter
  - hivemq-result
- concurrentUnit -> last released 2019, only in 1 project, test dependency => not an issue
- gradleOci-junitJupiter -> owned by the platform team, test dependency => not an issue
- jackson-databind-nullable -> last released Feb 2023, only in 1 project, test dependency => not an issue
- javet-sanitizer -> we are not on the latest version because of issues => not an issue
- jose4j -> only patch updates recently which are already not included in the patch updates due to the current config => not an issue
- mustache -> only patch updates since 2015 which are already not included in the patch updates due to the current config => not an issue
- openJdk-jol -> infrequent releases, last released Feb 2023, only in 1 project, test dependency => not an issue
- prometheus-simpleClient -> last released 2022, only in 1 project => not an issue
- reflections -> last released 2021, only in 1 poject, test dependency => not an issue
- zeroAllocationHashing -> very infrequent releases